### PR TITLE
Issue/536/advance game config

### LINKED ISSUE #536: Advance game configuration

### PR DESCRIPTION
**Pull request by Bedpaw**
closes #536

### CODE REVIEW

File: client/src/components/organisms/admin/game-advanced-rules.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <section>
+    <component  v-for="rule of advancedGameRules" :key="rule.ruleId"
+                :is="inputTypeComponent(rule.ruleType)"
+                v-model="rule.ruleValue"
+                :disabled="blockForm"
+                :assist="gameRulesUtils.getDescription(rule)"
+                v-bind="inputProps(rule)"
+                class="f-mb-2"
+    >
```
Comment (dbetka) [inline client/src/components/organisms/admin/game-advanced-rules.vue:10]: Should be like that (indents):
```vue
<component  
  v-for="rule of advancedGameRules" :key="rule.ruleId"
  :is="inputTypeComponent(rule.ruleType)"
  v-model="rule.ruleValue"
  :disabled="blockForm"
  :assist="gameRulesUtils.getDescription(rule)"
  v-bind="inputProps(rule)"
  class="f-mb-2"
>
```

File: client/src/components/atoms/checkbox.vue [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+        @focus="hasFocus = true"
+        @blur="hasFocus = false"
+        :id="id"
+        :disabled="isDisabled"
+      />
+      <a-icon
+        :name="iconName"
+        :type="ICONS_TYPES.outlined"
+        :class="{'f-text-primary': hasFocus}"
+      ></a-icon>
```
Comment (dbetka) [inline client/src/components/atoms/checkbox.vue:17]: Should be like below (tag close):
```vue
<a-icon
  :name="iconName"
  :type="ICONS_TYPES.outlined"
  :class="{'f-text-primary': hasFocus}"
/>
```

File: client/src/components/atoms/checkbox.vue [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+      <span class="f-pl-1 f-line-28">
+        <slot></slot>
```
Comment (dbetka) [inline client/src/components/atoms/checkbox.vue:19]:
```vue
<slot/>
```

File: client/src/config/event-config.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  mapRefreshTime: 15,
+  mapZoom: 2,
+  eventEndDate: null as DateType,
+  eventStartDate: null as DateType,
+  gameRules,
+  mapRefreshTimeOptions: [
+    {
+      label: '1 min',
+      value: secondsInMinute,
+    }, {
```
Comment (dbetka) [inline client/src/config/event-config.ts:38]:
```
}, {
```
to
```
}, 
{
```

File: client/src/utils/game-rules.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// Cannot use enum as object key
+const getOptionsEnumAsArray = (rule: GameRule) => getEnumValuesAsArray(rulesOptions[rule.ruleId]);
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// Cannot narrow type rule value, which depends from inputTypeEnum
+// Use only with rule.ruleType = InputTypeEnum.Select
+const getKeyFromOptions = (rule: GameRule) => getOptionsEnumAsArray(rule)[rule.ruleValue];
```
Comment (dbetka) [inline client/src/utils/game-rules.ts:18]: It should be without workaround like above. `ts-ignore` and `eslint-disable-next-line` is not safe solution.

File: client/src/dictionary/language/pl.js [hunk 6]
```diff
@@ -318,6 +322,35 @@ export const pl = {
...
     accountType: 'Typ konta',
   },
+  gameRules: {
+    category: {
+      user: {
+        map: 'Ustawienia mapy użytkownika',
+      },
+    },
+    description: {
+      GeolocationAvailability: 'NOT IMPLEMENTED',
```
Comment (dbetka) [inline client/src/dictionary/language/pl.js:332]: Translate `NOT IMPLEMENTED`

File: client/src/config/event-config.ts (hunk 4 above)
Comment (dbetka) [reply client/src/config/event-config.ts:38]: Or put it in one line:
```
mapRefreshTimeOptions: [
  { label: '1 min', value: secondsInMinute },
  { label: '5 min', value: 5 * secondsInMinute },
  ...
]
```

**Review by dbetka — CHANGES_REQUESTED** [review]
View looks good

File: client/src/components/atoms/checkbox.vue (hunk 2 above)
Comment (Bedpaw) [reply client/src/components/atoms/checkbox.vue]: done

File: client/src/config/event-config.ts (hunk 4 above)
Comment (Bedpaw) [reply client/src/config/event-config.ts]: ...or use [].map, done :smile:

File: client/src/dictionary/language/pl.js (hunk 6 above)
Comment (Bedpaw) [reply client/src/dictionary/language/pl.js]: Removed, we don't need this feature yet

File: client/src/utils/game-rules.ts (hunk 5 above)
Comment (Bedpaw) [reply client/src/utils/game-rules.ts]: done

File: client/src/components/atoms/checkbox.vue [hunk 7]
```diff
@@ -1,23 +1,28 @@
...
-  <label :for="id" :class="labelClass">
-    <input
-      class="a-checkbox"
-      type="checkbox"
-      v-model="vModel"
-      @focus="hasFocus = true"
-      @blur="hasFocus = false"
-      :id="id"
-      :disabled="isDisabled"
-    />
```
Comment (dbetka) [inline client/src/components/atoms/checkbox.vue:11]: Checkbox and text change color to green on focus. It look's strange.
Screenshot from 2021-11-22 19-04-02
Screenshot from 2021-11-22 19-03-56

Maybe change opacity on 0.76 - like buttons

File: client/src/components/organisms/admin/game-advanced-rules.vue [hunk 8]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <section>
+    <component
+      v-for="rule of advancedGameRules" :key="rule.ruleId"
```
Comment (dbetka) [inline client/src/components/organisms/admin/game-advanced-rules.vue:4]: Move `:key="rule.ruleId"` to next line

File: client/src/components/organisms/admin/game-advanced-rules.vue [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      :is="inputTypeComponent(rule.ruleType)"
+      v-model="rule.ruleValue"
+      :disabled="blockForm"
+      :assist="gameRulesUtils.getDescription(rule)"
+      v-bind="inputProps(rule)"
+      class="f-mb-2"
+    >
+      {{gameRulesUtils.getName(rule)}}
```
Comment (dbetka) [inline client/src/components/organisms/admin/game-advanced-rules.vue:12]: Add spaces before and after code in `{{ <code> }}`

File: client/src/router/utils.ts [hunk 10]
```diff
@@ -6,14 +6,14 @@ import { translator } from '../dictionary';
...
 export class AppRoute {
   public name: string; // !!!camelCase!!!
-  public path: string | null;
+  public path = ''
   public meta: EnterPermission;
   public icon = ICONS.success as string;
-  public label: string;
-  public shortLabel: string;
+  public label: string | null = null
+  public shortLabel: string | null = null
```
Comment (dbetka) [inline client/src/router/utils.ts:14]: Sometimes you're using semicolon and sometimes not. It should be standardized.

**Review by dbetka — CHANGES_REQUESTED** [review]

File: client/src/components/atoms/checkbox.vue (hunk 7 above)
Comment (Bedpaw) [reply client/src/components/atoms/checkbox.vue:11]: Moved to backlog

**Review by dbetka — APPROVED** [review]

**Pull request merged**